l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -602,15 +634,14 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
             // this message is a token withdrawal
 
             // Check that the message length is correct.
-            // It should be equal to the length of the function signature + address + address + uint256 = 4 + 20 + 20 + 32 =
-            // 76 (bytes).
-            require(_l2ToL1message.length == 76, "ShB wrong msg len 2");
-            (l1Receiver, offset) = UnsafeBytes.readAddress(_l2ToL1message, offset);
-            address l1Asset;
-            (l1Asset, offset) = UnsafeBytes.readAddress(_l2ToL1message, offset);
-            assetInfo = nativeTokenVault.getAssetInfo(l1Asset);
+            // It should be equal to the length of the function signature + address + uint256 + address = 4 + 32 + 32 + 32 =
+            // 100 (bytes).
+            require(_l2ToL1message.length == 100, "ShB wrong msg len 2");
```

Comment:
I don't think we will know the length of the message. In particular we need to support dynamic amount of data for custom bridges, that is not here yet

Reply:
We can store the mapping of vault -> message length as an option

Reply:
Can't we just expect the l2Message to be of some format and abi decode it?

Reply:
We can add the type to _l2ToL1message and parse accordingly if we expect the messages to be of varying size or encoding.

Let's sync on this one so it's faster

Reply:
Agree with Kalman, not sure why we need to limit the structure of the assetdata for withdrawals

Reply:
I think the simply way will be to have encoding of assetInfo (bytes32) + assetData (bytes memory), which we simply pass to the respective assetHandler without checks, so it's flexible and up to the correct assetHandler to decode

Reply:
done